styleguide/drafts/tooling.rst

```diff
@@ -0,0 +1,518 @@
+    * - Simple to write tests
+      -
+    * - Compatible with built in unittest module
+      -
+    * - Compatible with nose
+      -
+    * - Powerful and easy fixture(s) mechanism
+      -
+    * - Parameterized tests
+      -
+    * - Grouping and marking of tests for different executions
+      -
+    * - Supports pyproject.toml
+      -
+    * - Extensible through plugin mechanism
+      -
+
+Alternatives
+~~~~~~~~~~~~
+* pyunit_
+* nose_
+
+Documentation
+_____________
+
+What to say... we all want to have it and read it if we need it, but most of us don't want to write it.
+
+.. figure:: https://imgs.xkcd.com/comics/manuals.png
+   :alt: xkcd-1343
+   :target: https://xkcd.com/1343/
+
+   source: `xkcd.com <xkcd_>`_
+   license: `CC BY-NC 2.5`_
+
+
+Sphinx_
++++++++
+`Project Homepage <Sphinx_www_>`_
+
+Sphinx is widely used within the python space and outside of it, because it is very powerful.
+Admittedly it's powerfulness comes with a cost of complexity for bigger setups.
+Still the overall the benefits outweigh the cost, and with the detail in mind that
+lots of projects are using sphinx it is worth the effort to learn and use it.
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 50 50
+
+    * - Pros ✅
+      - Cons ❌
+    * - Widely used
+      - Additional dependency
+    * - Lots of extensions
+      - Restructured text can be quirky at times
+    * - Lots of output formats are supported
+      - Lots to learn/know regarding .rst and sphinx
+    * - Basics are easy to learn
+      -
+    * - Extensions for api documentation
+      -
+    * - Plugin mechanism
+      -
+
+Alternatives
+~~~~~~~~~~~~
+
+* `mkdocs <https://github.com/mkdocs/mkdocs>`_
+* `pydoc <https://pdoc.dev/docs/pdoc.html>`_
+
+
+Dependencies & Packaging
+________________________
+
+.. figure:: https://imgs.xkcd.com/comics/dependency.png
+   :alt: xkcd-2347
+   :target: https://xkcd.com/2347/
+
+   source: `xkcd.com <xkcd_>`_
+   license: `CC BY-NC 2.5`_
+
+
+Poetry_
++++++++
+`Project Homepage <Poetry_www_>`_
+
+Poetry has a very good package/version resolver and simplifies packaging and updating dependencies significantly.
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 50 50
+
+    * - Pros ✅
+      - Cons ❌
+    * - Good dependency resolver
+      - Fairly new, some edge cases my not supported yet
+    * - Uses pyproject.toml for configuration
+      - Toml file definition is less flexible than setup.py based one
+    * - Takes care of project versioning
+      - Editable and/or repo based install(s) are not possible "out of the box"?
+    * - Dependency pinning
+      - setup.py is still required for installations with older versions of pip
+    * - Very active development & community
+      -
+    * - Good & Powerful CLI
+      -
+    * - No manual dependency editing required
+      -
+
+Alternatives
+~~~~~~~~~~~~
+
+* `pipenv <https://pipenv.pypa.io/en/latest/>`_
```

Comment:
should we explain, why we don't use pipenv? I have two reasons. First, the project was more than a year dead and no maintainer, currently it has one, but that wasn't a good sign and second, the resolver was not as powerful.

Reply:
We definitely wanna do explain why we have taken which choice in the final guide.